ocaml-lsp-server/src/import.ml

```diff
@@ -2,7 +2,82 @@
    listed alphabetically. Try to keep the order. *)
 include struct
   open Stdune
-  module Array = Array
+
+  module Array = struct
+    include Array
+
+    module Sub : sig
+      type 'a t
+
+      val make : 'a array -> start:int -> ?length:int -> unit -> 'a t
+
+      val get : 'a t -> int -> 'a
+
+      val set : 'a t -> int -> 'a -> unit
+
+      val is_empty : 'a t -> bool
+
+      val length : 'a t -> int
+
+      (** compute length of common suffix between two sub-arrays *)
+      val common_suffix_len : 'a t -> 'a t -> int
+    end = struct
+      type 'a t =
+        { arr : 'a array
+        ; start : int
+        ; end_excl : int
+        }
+
+      let make arr ~start ?length () =
+        let length = Option.value length ~default:(Array.length arr - start) in
+        { arr; start; end_excl = start + length }
+
+      let offset_index t i =
+        let ix = t.start + i in
+        if ix < t.end_excl then ix
+        else invalid_arg "subarray index out of bounds"
+
+      let get t i = t.arr.(offset_index t i)
+
+      let set t i x = t.arr.(offset_index t i) <- x
+
+      let length t = t.end_excl - t.start
+
+      let is_empty t = length t = 0
+
+      let common_suffix_len ai aj =
+        if length ai = 0 || length aj = 0 then 0
+        else
+          let i = ref (length ai - 1) in
+          let j = ref (length aj - 1) in
+          while !i >= 0 && !j >= 0 && get ai !i = get aj !j do
+            decr i;
+            decr j
+          done;
+          length ai - !i - 1
+    end
+
+    let split_into_subs arr ~at =
+      if at < 0 then
+        invalid_arg "split_into_subs: [~at] argument must be non-negative";
+      if at < Array.length arr then
+        let left = Sub.make arr ~start:0 ~length:at () in
+        let right = Sub.make arr ~start:at () in
+        (left, right)
+      else invalid_arg "split_into_subs: [~at] argument out of bounds of array"
+
+    let common_prefix_len (a : 'a array) (b : 'a array) : int =
+      let i = ref 0 in
+      while
+        !i < Array.length a
+        && !i < Array.length b
```

Comment:
A little cleaner would be to compute `min (length a) (length b)` and only compared it to that.